New recipe for tardis.

tesujimath opened the pull request:
* [X] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [X] This PR adds a new recipe.
* [X] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

daler commented on recipes/tardis/meta.yaml:
```diff
@@ -0,0 +1,64 @@
+
+requirements:
+  build:
+    - python 2.7.*
```
Change to `- python

tesujimath replied:
Done.

daler commented on recipes/tardis/meta.yaml:
```diff
@@ -0,0 +1,64 @@
+  sha256: {{ sha256 }}
+
+build:
+  skip: True  # [win]
```
We don't build on windows, but it looks like you want to restrict to py27, so change `# [win]` to `# [not py27]`

tesujimath replied:
Thanks, yes.  Done.

daler commented on recipes/tardis/meta.yaml:
```diff
@@ -0,0 +1,64 @@
+    - zlib {{CONDA_ZLIB}}*
+
+  run:
+    - python 2.7.*
```
change to `- python`

tesujimath replied:
Done

daler commented on recipes/tardis/meta.yaml:
```diff
@@ -0,0 +1,64 @@
+    - kseq_split/test.fastq
+  commands:
+    - tardis -h
+    - kseq_split -s 0.1 -o fastq test.fastq 10 split-%05d.fastq
```
I'm surprised this works . . . @bgruening does mulled-build copy over test files from source now?

tesujimath replied:
Ah, I see kseq_split is returning a success exit code even when it can't find its input files.  Fixed now.

tesujimath commented:
@daler Thanks for the review.  I have addressed your comments, and pushed a new version.  Tardis absolutely is bioinformatics specific.  It does domain-aware job-splitting and output file recombining, for one thing.

daler commented:
@tesujimath thanks. I had to read through more of the readme to find the bioinformatics-specific stuff; for others following this, https://github.com/AgResearch/tardis says:

> - content-aware splitting of large input files. tardis includes a fast C based fasta and fastq splitter
> - handling multiple inputs which must be split in lockstep - for example pairs of paired-end sequence files

I also see:

> - there are a couple of features such as filtering sequences by length that require biopython - however for most use-cases tardis will run fine without biopython

Do you think it's worth it to include biopython as a dependency?

tesujimath commented:
@daler We really don't want to include BioPython as a dependency.  Tardis works find without it.  It only uses it for one corner case, which is quite rare.

Updating the README is in hand.

daler commented:
OK, thank you @tesujimath!
